src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/BasmClientApiService.kt

```diff
@@ -40,13 +48,78 @@ class BasmClientApiService(
         .block(basmApiTimeout)
     }
       .onFailure { recordLocationLookupFailure(it) }
-      .onSuccess { recordIfLocationNotFound(it) }.getOrNull()?.data?.elementAtOrNull(0)?.name()?.toUpperCase()
+      .onSuccess { recordIfLocationNotFound(it) }.getOrNull()?.locations?.elementAtOrNull(0)?.name?.toUpperCase()
+  }
+
+  fun findNomisAgenciesCreatedOn(date: LocalDate): List<BasmNomisLocation> {
+    fun recordFailure(error: Throwable) {
+      logger.error("An error occurred trying to find location by date '$date' on calling BaSM API: ${error.message}")
+      monitoringService.capture("An error occurred trying to find location by date '$date' on calling BaSM API: ${error.message}")
+    }
+
+    return Result.runCatching {
+      basmApiWebClient
+        .get()
+        .uri("api/reference/locations?filter[created_at]=$date")
+        .retrieve()
+        .bodyToMono(LocationResponse::class.java)
+        .map { it.locations }
+        .block(basmApiTimeout)
+    }
+      .onFailure { recordFailure(it) }
+      .getOrDefault(listOf()).filterNotNull()
   }
 }
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
-private data class LocationResponse(val data: List<Location> = listOf())
+data class LocationResponse(
+  @JsonProperty("data")
+  @JsonDeserialize(contentUsing = NomisLocationDeserializer::class)
+  val locations: List<BasmNomisLocation> = listOf()
+)
+
+data class BasmNomisLocation(
+  val name: String,
+  val agencyId: String,
+  val locationType: LocationType,
+  val createdAt: LocalDate
+)
+
+object NomisLocationDeserializer : JsonDeserializer<BasmNomisLocation>() {
+
+  private val mayBe: Map<String, LocationType> = mapOf(
+    "approved_premises" to LocationType.APP,
```

Comment:
These are the support location types returned from BaSM.  Courts are not in here they have to be treated differently due to not providing an actual type for each court type.  Courts are handled a bit further down.